pandas/_libs/lib.pyi

```diff
@@ -51,23 +53,65 @@ def is_float_array(values: np.ndarray, skipna: bool = False): ...
 def is_integer_array(values: np.ndarray, skipna: bool = False): ...
 def is_bool_array(values: np.ndarray, skipna: bool = False): ...
 
-def fast_multiget(mapping: dict, keys: np.ndarray, default=np.nan) -> ArrayLike: ...
+def fast_multiget(mapping: dict, keys: np.ndarray, default=np.nan) -> np.ndarray: ...
 
 def fast_unique_multiple_list_gen(gen: Generator, sort: bool = True) -> list: ...
 def fast_unique_multiple_list(lists: list, sort: bool = True) -> list: ...
 def fast_unique_multiple(arrays: list, sort: bool = True) -> list: ...
 
 def map_infer(
     arr: np.ndarray, f: Callable[[Any], Any], convert: bool = True, ignore_na: bool = False
+) -> np.ndarray: ...
+
+
+@overload  # both convert_datetime and convert_to_nullable_integer False -> np.ndarray
+def maybe_convert_objects(
+    objects: np.ndarray,  # np.ndarray[object]
+    try_float: bool = ...,
+    safe: bool = ...,
+    convert_datetime: Literal[False] = ...,
+    convert_timedelta: bool = ...,
+    convert_to_nullable_integer: Literal[False] = ...,
+) -> np.ndarray: ...
+
+@overload
+def maybe_convert_objects(
+    objects: np.ndarray,  # np.ndarray[object]
+    try_float: bool = ...,
+    safe: bool = ...,
+    convert_datetime: Literal[False] = False,
+    convert_timedelta: bool = ...,
+    convert_to_nullable_integer: Literal[True] = ...,
 ) -> ArrayLike: ...
```

Comment:
Nice...the reason the overload works here (with both `Literal[True]` and `Literal[False]` taking defaults) is that the return type of the second one is a supertype of the return type of the first one. Haven't tried running this yet, but it looks correct to me - if you don't specify `convert_to_nullable_integer`, it should match the first overload

Reply:
im not clear on what you're asking for here

Reply:
nothing :) I was just commenting in case it wasn't clear why here you were able to let both overload variants take defaults but in most other cases we could only let one of the two take defaults